keyboards/hidtech/bastyl/keymaps/nstickney/keymap.c

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2020 Quentin LEBASTARD <qlebastard@gmail.com>
+ * Copyright 2020 Anthony MARIN <anthony@hidtech.ca>
+ * Copyright 2021 Stick <stick@stma.is>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http:www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
```

Comment:
Technically, because you have this included in the nstickney.h file, you don't need it here.  But it does no harm here, either. 

```suggestion
```

Reply:
That fails...

```
QMK Firmware 0.11.40
WARNING:
 Can not run bin/qmk! This tool will be required when the develop branch is merged on 2020 Aug 29.

 Please run util/qmk_install.sh to install all the dependencies QMK requires.

Making keebio/iris/rev2 with keymap nstickney and target dfu

avr-gcc (GCC) 10.2.0
Copyright (C) 2020 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

Size before:
   text	   data	    bss	    dec	    hex	filename
      0	  24172	      0	  24172	   5e6c	.build/keebio_iris_rev2_nstickney.hex

Compiling: users/nstickney/nstickney.c                                                             users/nstickney/nstickney.c:18:18: error: unknown type name 'qk_tap_dance_state_t'
   18 | void dance_layer(qk_tap_dance_state_t *state, void *user_data) {
      |                  ^~~~~~~~~~~~~~~~~~~~
users/nstickney/nstickney.c:34:26: error: unknown type name 'qk_tap_dance_state_t'
   34 | void dance_lock_finished(qk_tap_dance_state_t *state, void *user_data) {
      |                          ^~~~~~~~~~~~~~~~~~~~
users/nstickney/nstickney.c:53:23: error: unknown type name 'qk_tap_dance_state_t'
   53 | void dance_lock_reset(qk_tap_dance_state_t *state, void *user_data) {
      |                       ^~~~~~~~~~~~~~~~~~~~
users/nstickney/nstickney.c:72:1: error: unknown type name 'qk_tap_dance_action_t'
   72 | qk_tap_dance_action_t tap_dance_actions[] = {
      | ^~~~~~~~~~~~~~~~~~~~~
users/nstickney/nstickney.c:73:3: error: 'LOCKS' undeclared here (not in a function)
   73 |  [LOCKS] = ACTION_TAP_DANCE_FN_ADVANCED(NULL, dance_lock_finished, dance_lock_reset),
      |   ^~~~~
users/nstickney/nstickney.c:73:3: error: array index in initializer not of integer type
users/nstickney/nstickney.c:73:3: note: (near initialization for 'tap_dance_actions')
users/nstickney/nstickney.c:73:12: error: implicit declaration of function 'ACTION_TAP_DANCE_FN_ADVANCED' [-Werror=implicit-function-declaration]
   73 |  [LOCKS] = ACTION_TAP_DANCE_FN_ADVANCED(NULL, dance_lock_finished, dance_lock_reset),
      |            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
users/nstickney/nstickney.c:73:41: error: 'NULL' undeclared here (not in a function)
   73 |  [LOCKS] = ACTION_TAP_DANCE_FN_ADVANCED(NULL, dance_lock_finished, dance_lock_reset),
      |                                         ^~~~
users/nstickney/nstickney.c:1:1: note: 'NULL' is defined in header '<stddef.h>'; did you forget to '#include <stddef.h>'?
  +++ |+#include <stddef.h>
    1 | /* Copyright 2021 @nstickney
users/nstickney/nstickney.c:73:47: error: 'dance_lock_finished' undeclared here (not in a function)
   73 |  [LOCKS] = ACTION_TAP_DANCE_FN_ADVANCED(NULL, dance_lock_finished, dance_lock_reset),
      |                                               ^~~~~~~~~~~~~~~~~~~
users/nstickney/nstickney.c:73:68: error: 'dance_lock_reset' undeclared here (not in a function)
   73 |  [LOCKS] = ACTION_TAP_DANCE_FN_ADVANCED(NULL, dance_lock_finished, dance_lock_reset),
      |                                                                    ^~~~~~~~~~~~~~~~
users/nstickney/nstickney.c:74:3: error: 'LAYERS' undeclared here (not in a function)
   74 |  [LAYERS] = ACTION_TAP_DANCE_FN(dance_layer)
      |   ^~~~~~
users/nstickney/nstickney.c:74:3: error: array index in initializer not of integer type
users/nstickney/nstickney.c:74:3: note: (near initialization for 'tap_dance_actions')
users/nstickney/nstickney.c:74:13: error: implicit declaration of function 'ACTION_TAP_DANCE_F ' [-Werror=implicit-function-declaration]
   74 |  [LAYERS] = ACTION_TAP_DANCE_FN(dance_layer)
      |             ^~~~~~~~~~~~~~~~~~~
users/nstickney/nstickney.c:74:33: error: 'dance_layer' undeclared here (not in a function)
   74 |  [LAYERS] = ACTION_TAP_DANCE_FN(dance_layer)
      |                                 ^~~~~~~~~~~
users/nstickney/nstickney.c: In function 'keyboard_post_init_user':
users/nstickney/nstickney.c:82:5: error: implicit declaration of function 'rgblight_enable_noeeprom' [-Werror=implicit-function-declaration]
   82 |     rgblight_enable_noeeprom();
      |     ^~~~~~~~~~~~~~~~~~~~~~~~
users/nstickney/nstickney.c:83:5: error: implicit declaration of function 'rgblight_mode_noeeprom' [-Werror=implicit-function-declaration]
   83 |     rgblight_mode_noeeprom(RGBLIGHT_MODE_STATIC_LIGHT);
      |     ^~~~~~~~~~~~~~~~~~~~~~
users/nstickney/nstickney.c:83:28: error: 'RGBLIGHT_MODE_STATIC_LIGHT' undeclared (first use in this function)
   83 |     rgblight_mode_noeeprom(RGBLIGHT_MODE_STATIC_LIGHT);
      |                            ^~~~~~~~~~~~~~~~~~~~~~~~~~
users/nstickney/nstickney.c:83:28: note: each undeclared identifier is reported only once for each function it appears in
users/nstickney/nstickney.c:85:9: error: implicit declaration of function 'rgblight_sethsv_noeeprom' [-Werror=implicit-function-declaration]
   85 |         rgblight_sethsv_noeeprom((i + LAYER_HUE[BASE]) % 256, 255, 136);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~
users/nstickney/nstickney.c:85:49: error: 'BASE' undeclared (first use in this function); did you mean 'SE'?
   85 |         rgblight_sethsv_noeeprom((i + LAYER_HUE[BASE]) % 256, 255, 136);
      |                                                 ^~~~
      |                                                 SE
users/nstickney/nstickney.c:85:48: error: array subscript is not an integer
   85 |         rgblight_sethsv_noeeprom((i + LAYER_HUE[BASE]) % 256, 255, 136);
      |                                                ^
users/nstickney/nstickney.c:85:56: error: invalid operands to binary % (have 'int *' and 'int')
   85 |         rgblight_sethsv_noeeprom((i + LAYER_HUE[BASE]) % 256, 255, 136);
      |                                  ~~~~~~~~~~~~~~~~~~~~~ ^
      |                                     |
      |                                     int *
users/nstickney/nstickney.c:86:9: error: implicit declaration of function 'wait_ms' [-Werror=implicit-function-declaration]
   86 |         wait_ms(8);
      |         ^~~~~~~
users/nstickney/nstickney.c: At top level:
users/nstickney/nstickney.c:91:1: error: unknown type name 'layer_state_t'
   91 | layer_state_t layer_state_set_user(layer_state_t state) {
      | ^~~~~~~~~~~~~
users/nstickney/nstickney.c:91:36: error: unknown type name 'layer_state_t'
   91 | layer_state_t layer_state_set_user(layer_state_t state) {
      |                                    ^~~~~~~~~~~~~
cc1: all warnings being treated as errors
 [ERRORS]
 |
 |
 |
make[1]: *** [tmk_core/rules.mk:412: .build/obj_keebio_iris_rev2_nstickney/nstickney.o] Error 1
Make finished with errors
make: *** [Makefile:523: keebio/iris/rev2:nstickney:dfu] Error 1
```